### ISSUE TITLE
Weird node crash error

### ISSUE THREAD

**Issue by SteveALee** [Azure/azure-webjobs-sdk-script#836]
@christopheranderson wrote this issue from the conference floor, so blame him if it's not clear enough...
# Code
## index.js

```
const xs = require('xstream').default
const run = require('@cycle/xstream-run').run
const makeHTTPDriver = require('@cycle/http').makeHTTPDriver
const makeFunctionsDriver = require('./functionsDriver.js')

function main(sources) {
    const log = sources.FaaS.log

    const input$ = sources.FaaS
        .map((s) => `context.req.originalUrl: ${s.context.req.originalUrl} /
                        inArgs[0]: ${JSON.stringify(s.inArgs[0])}`)

    const ticks$ = xs.periodic(1000)
        .map(t => (t + 1) * 100)

    const log$ = xs.merge(input$, ticks$)

    const trigger$ = ticks$
        .drop(2)
        .debug((i) => log(`trigger tick:`, `${i}`))

    const request$ = trigger$
        .map(() => ({
                        url: 'http://jsonplaceholder.typicode.com/users/1',
                        method: 'GET'
                    }))

    const response$ = sources.HTTP
        .select()
        .flatten()
    const user1Data$ = response$.map(response => response.body)
    const exit$ = user1Data$
        .map(u => ({ status: 200, body: u}))

    return {
        FaaS: exit$,
        log: log$,
        HTTP: request$
    }
}

module.exports = function (context, ...inArgs) {
    const dispwrap = {}             // because JS doesn't do pass-by-ref
    const functionsDriver = makeFunctionsDriver(context, inArgs, dispwrap)
    const log = functionsDriver.log

    const drivers = {
        FaaS: functionsDriver,
        HTTP: makeHTTPDriver(),
        log: msg$ => { msg$.addListener({next: msg => log(`${msg}`) }) }
    }

    dispwrap.disposer = run(main, drivers)
}
```
# function.json

```
{
  "bindings": [
    {
      "type": "httpTrigger",
      "direction": "in",
      "webHookType": "genericJson",
      "name": "req"
    },
    {
      "type": "http",
      "direction": "out",
      "name": "res"
    }
  ],
  "disabled": false
}

```
## functionsDriver.js

```
const xs = require('xstream').default

module.exports = function makeFunctionsDriver(context, inArgs, dispwrap) {
    const log = (...a) => context.log(...a)

    function driver(s$) {
        s$.addListener({ next: (i) => { setTimeout(() => {  // next tick to allow other listeners to be iterated
                                            context.res = i
                                            if (dispwrap &&
                                                typeof dispwrap.disposer === 'function') {
                                                dispwrap.disposer()
                                            }
                                            context.done()
                                            }, 1)
                         },
                         error: (e) => { context.log(e) },
                         complete: () => {}
        })

        const stream = xs.createWithMemory({
            start: listener => {
                setTimeout(() => {listener.next({context, inArgs})}, 1)
            },
            stop: () => {},
        })

        stream.log = log
        return stream
    }

    driver.log = log
    return driver
}
```

Stack trace:

```
  Function had errors. See Azure WebJobs SDK dashboard for details. Instance ID is 'f6033f52-c8c6-4b1c-a8d3-f04bc65b83ca'
Microsoft.Azure.WebJobs.Host.FunctionInvocationException: Exception while executing function: Functions.cycle-serverless ---> System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Microsoft.Azure.WebJobs.Script.Description.NodeFunctionInvoker.CreateScriptExecutionContext(Object input, DataType dataType, TraceWriter traceWriter, FunctionInvocationContext invocationContext)
   at Microsoft.Azure.WebJobs.Script.Description.NodeFunctionInvoker.<InvokeCore>d__24.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.Azure.WebJobs.Script.Description.FunctionInvokerBase.<Invoke>d__23.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker`1.<InvokeAsync>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<InvokeAsync>d__52.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithWatchersAsync>d__3a.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithLoggingAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithLoggingAsync>d__1a.MoveNext()
   --- End of inner exception stack trace ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithLoggingAsync>d__1a.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<TryExecuteAsync>d__2.MoveNext()
```

**Comment by mamaso** [comment]
Can't repro this, and I get what looks like the expected output:

![image](https://cloud.githubusercontent.com/assets/9870694/19865778/d2e2811e-9f5a-11e6-85c5-cc1eb87b3212.png)

**Comment by mamaso** [comment]
@christopheranderson if you want me to dig in deeper add me as a coadmin on the account you're seeing this

**Comment by mamaso** [comment]
actually, this looks like #849

**Comment by SteveALee** [comment]
@christopheranderson @mamaso I thought I'd take another looks at this but cannot find the instructions to run locally. Can you provide a pointer please? Thanks

**Comment by SteveALee** [comment]
> actually, this looks like it might be #849

@mamaso you may be correct. If I recall with triggered it form the Azure Functions control panel and might have just deleted the body. Once I get the instructions I'll look at it again.

**Comment by SteveALee** [comment]
Not sure if this matters at all but my Azure Tools for VS Code extension was on 0.0.5 and no update offered in the list of extensions to the latest 0.0.9. I manually uninstalled and installed latest.

**Comment by mamaso** [comment]
Hey @SteveALee, try using the [azure functions cli](https://www.npmjs.com/package/azurefunctions) if you want to run locally.

You can also repro via the functions portal if you like.

**Comment by christopheranderson** [comment]
Labeling this as no-repro until we know the root cause

**Comment by SteveALee** [comment]
> You can also repro via the functions portal if you like.

I tried but the request body always shows a min of 1 line so may not be empty. Works fine. I just updated Functions on the portal to the latest. 

I tried using the local functions cli but need to become more familiar with it and not sure how yet

I think I must be doing somehting wrong as a ```func list`` gives this error

Error: Object reference not set to an instance of an object.

I tried in both the project folder and child function folder.

**Comment by mamaso** [comment]
So here's what I do from the portal to throw the KeyNotFoundException:

![image](https://cloud.githubusercontent.com/assets/9870694/19979175/c66e5738-a1b6-11e6-9b17-56c5c2c63453.png)

**Comment by SteveALee** [comment]
@mamaso well now we have switched places because I cannot repo it. You did exactly what I did AFAICS. We have to assume the body is empty even though line 1 exists. Thats probably just an artifact so you can enter data.

So what could be different?
* Runtime version: latest (~0.9)
* The source could be different - I will check
* npm module - npm is non deterministic but unlikely to be the issue here
* I do not understand what this "query" is. I have no UI to remove it.

![untitled](https://cloud.githubusercontent.com/assets/618922/19979830/c15fe7e2-a1f3-11e6-9cc9-4888ed2378ee.png)

**Comment by SteveALee** [comment]
d-oh! code is the required API security value in the query string of the URL. SO it makes sense it is there. I can see it in the function URL but can't edit it their either. So we can ignore that.

**Comment by mamaso** [comment]
Ah, nvmd, I see we got the 0.9 release out, I was still running 0.8.  Updated and ran without issue.

0.9 has the fix for #849, so I think we can say this has been resolved.

**Comment by SteveALee** [comment]
\0/ excellent.
